ansible/roles/csm.rr.mgmt_nodes_placement_validation/files/mgmt_nodes_placement_validation.py

```diff
@@ -0,0 +1,257 @@
+
+    if master_nodes_cnt >= min_master_node_cnt:
+        if rack_cnt == min_rack_cnt and master_nodes_cnt == min_master_node_cnt and  missing_cnt >= node_rack_missing_quorum:
+            print(bcolors.FAIL + "\nOne or more management racks missing master node under minimum 3 rack requirement for RR" + bcolors.ENDC)
+            print(bcolors.FAIL + "\nPlease re arrange the master nodes for equal distribution" + bcolors.ENDC)
+            print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+            sys.exit(1)
+        elif rack_cnt > min_rack_cnt and master_nodes_cnt >= min_master_node_cnt and missing_cnt > node_rack_missing_quorum:
+            print(bcolors.FAIL + "\nMore than one rack missing master node in a", rack_cnt , "rack system" + bcolors.ENDC)
+            print(bcolors.FAIL + "\nPlease re arrange the master nodes for equal distribution" + bcolors.ENDC)
+            print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+            sys.exit(1)
+    else:
+        print(bcolors.FAIL + "\nRack Resiliency can not be enabled with", master_nodes_cnt, "master nodes" + bcolors.ENDC)
+        print(bcolors.FAIL + "\nNeed minimum of 3 master nodes configured" + bcolors.ENDC)
+        print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+        sys.exit(1)
+
+def validate_worker_nodes_placement(placements_dict):
+    rack_cnt=1
+    worker_nodes_cnt=0
+    missing_cnt=0
+
+    print(bcolors.OKGREEN + bcolors.UNDERLINE + "\nValidate Management Worker nodes"+ bcolors.ENDC)
+
+    for rack_cnt, (rack_id, nodes) in enumerate(placements_dict.items()):
+        print("\nRack number: ", rack_cnt)
+        print("Rack ID: ", rack_id)
+        print("Management Nodes: ", nodes)
+
+        found=0
+        for node in nodes:
+            if re.match(r"^.*ncn-w00[0-9]$", node):
+                # Found worker node in a rack
+                worker_nodes_cnt+=1
+                found=1
+
+        if found == 0:
+            # Missing worker node in a rack
+            missing=1
+            missing_cnt+=1
+
+        rack_cnt+=1
+
+    print("\nTotal number of worker nodes: ", worker_nodes_cnt)
+
+    node_rack_missing_quorum=get_node_rack_missing_quorum(rack_cnt)
+
+    if worker_nodes_cnt >= min_worker_node_cnt:
+        if rack_cnt >= min_rack_cnt and worker_nodes_cnt >= min_worker_node_cnt and  missing_cnt > node_rack_missing_quorum:
+            if rack_cnt == min_rack_cnt and worker_nodes_cnt == min_worker_node_cnt
+                print(bcolors.FAIL + "\nOne or more racks missing worker node under minimum 3 rack condition" + bcolors.ENDC)
+                print(bcolors.FAIL + "\nPlease re arrange the worker nodes for equal distribution" + bcolors.ENDC)
+                print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+            elif rack_cnt > min_rack_cnt and worker_nodes_cnt >= min_worker_node_cnt
```

Comment:
There are colons missing at the end of two of these lines, which are Python syntax errors. `pylint` would have caught these errors. Please verify that all Python code has no `pylint` errors before re-requesting review.
```suggestion
            if rack_cnt == min_rack_cnt and worker_nodes_cnt == min_worker_node_cnt:
                print(bcolors.FAIL + "\nOne or more racks missing worker node under minimum 3 rack condition" + bcolors.ENDC)
                print(bcolors.FAIL + "\nPlease re arrange the worker nodes for equal distribution" + bcolors.ENDC)
                print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
            elif rack_cnt > min_rack_cnt and worker_nodes_cnt >= min_worker_node_cnt:
```